pkg/core/diagnostics.go

```diff
@@ -8,17 +8,28 @@ import (
 	"github.com/spf13/viper"
 )
 
+type DiagnosticsSummary struct {
+	Total      int            `json:"total,omitempty"`
+	ByType     map[string]int `json:"by_type,omitempty"`
+	BySeverity map[string]int `json:"by_severity,omitempty"`
+}
+
 type SentryDiagnostic struct {
 	diag.Diagnostic
 
 	Tags   map[string]string
 	Ignore bool
 }
 
-type DiagnosticsSummary struct {
-	Total      int            `json:"total,omitempty"`
-	ByType     map[string]int `json:"by_type,omitempty"`
-	BySeverity map[string]int `json:"by_severity,omitempty"`
+func (d SentryDiagnostic) Redacted() diag.Diagnostic {
+	v, ok := d.Diagnostic.(diag.Redactable)
+	if !ok {
+		return d
+	}
+	if r := v.Redacted(); r != nil {
+		return r
```

Comment:
Shouldn't this create and return a `SentryDiagnostic` to keep the "enrichments" in the redacted diag as well?

Reply:
Not sure, that's why I asked you for the review :)

It was hard enough for me to track the issue down across everything that happens around diags. I'lll update, thanks!